Libs/MRML/Core/vtkMRMLPlotLayoutNode.h

```diff
@@ -0,0 +1,261 @@
+
+  Copyright (c) Kapteyn Astronomical Institute
+  University of Groningen, Groningen, Netherlands. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Davide Punzo, Kapteyn Astronomical Institute,
+  and was supported through the European Research Consil grant nr. 291531.
+
+==============================================================================*/
+
+#ifndef __vtkMRMLPlotLayoutNode_h
+#define __vtkMRMLPlotLayoutNode_h
+
+#include "vtkMRMLNode.h"
+
+class vtkCollection;
+class vtkDataObject;
+class vtkMRMLPlotNode;
+class vtkStringArray;
+
+class PlotIDMap;
+class PlotLayoutPropertyMap;
+
+#include <string>
+
+/// \brief MRML node for referencing a collection of data to plot.
+class VTK_MRML_EXPORT vtkMRMLPlotLayoutNode : public vtkMRMLNode
+{
+ public:
+  //----------------------------------------------------------------
+  /// Standard methods for MRML nodes
+  //----------------------------------------------------------------
+
+  static vtkMRMLPlotLayoutNode *New();
+  vtkTypeMacro(vtkMRMLPlotLayoutNode,vtkMRMLNode);
+
+  void PrintSelf(ostream& os, vtkIndent indent);
+
+  virtual vtkMRMLNode* CreateNodeInstance();
+
+  ///
+  /// Set node attributes
+  virtual void ReadXMLAttributes( const char** atts);
+
+  ///
+  /// Write this node's information to a MRML file in XML format.
+  virtual void WriteXML(ostream& of, int indent);
+
+  ///
+  /// Copy the node's attributes to this object
+  virtual void Copy(vtkMRMLNode *node);
+
+  ///
+  /// Get node XML tag name (like Volume, Model)
+  virtual const char* GetNodeTagName() {return "PlotLayout";};
+
+  ///
+  /// Method to propagate events generated in mrml
+  virtual void ProcessMRMLEvents ( vtkObject *caller, unsigned long event, void *callData );
+
+  //----------------------------------------------------------------
+  /// Access methods
+  //----------------------------------------------------------------
+
+  ///
+  /// Add and Observe a Plot to the PlotLayout. Parameter "name" is used for
+  /// referencing the Plot when setting properties for plotting the
+  /// Plot or for removing the Plot from the PlotLayout.
+  virtual void AddAndObservePlot(const char *name, const char * id);
+
+  ///
+  /// Remove a Plot and Observation from the PlotLayout by a particular name
+  virtual void RemovePlotAndObservationByName(const char *name);
+
+  ///
+  /// Remove a Plot and Observation from the PlotLayout by ID
+  virtual void RemovePlotAndObservationByID(const char *id);
+
+  ///
+  /// Update all missing Observations
```

Comment:
This most likley will disapper after I use properly the reference infrastracture. Essentially this method check if all the PlotNode are observeing vtkplot.